reconcilers/reconcilers.go

```diff
@@ -146,15 +149,19 @@ func (r *ParentReconciler) reconcile(ctx context.Context, parent client.Object)
 	}
 
 	r.copyGeneration(parent)
-
 	return result, nil
 }
+func (r *ParentReconciler) isStatusSet(obj client.Object) bool {
```

Comment:
We don't really care if status is "set" or not, we care if the resource has status.

```suggestion
func (r *ParentReconciler) hasStatus(obj client.Object) bool {
```